File: backend/src/BookingManager.Api/Models/Booking.cs
namespace BookingManager.Api.Models;

public class Booking
{
    public Guid Id { get; set; }
    public Guid ResourceId { get; set; }
    public Guid UserId { get; set; }
    public DateTime StartDateTime { get; set; }
    public DateTime EndDateTime { get; set; }
    public BookingStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? CancelledAt { get; set; }
    public Guid? CancelledBy { get; set; }

    public Resource? Resource { get; set; }
    public User? User { get; set; }
}

File: frontend/booking-manager-ui/lib/types/booking.ts
export type BookingStatus = "Confirmed" | "Cancelled";

export interface Booking {
  id: string;
  resourceId: string;
  userId: string;
  startDateTime: string;
  endDateTime: string;
  status: BookingStatus;
  createdAt: string;
  cancelledAt: string | null;
}

export interface CreateBookingRequest {
  resourceId: string;
  userId: string;
  startDateTime: string;
  endDateTime: string;
}
